pkg/registry/core/service/storage/rest.go

```diff
@@ -599,10 +599,12 @@ func isValidAddress(ctx context.Context, addr *api.EndpointAddress, pods rest.Ge
 	if pod == nil {
 		return fmt.Errorf("pod is missing, skipping (%s/%s)", addr.TargetRef.Namespace, addr.TargetRef.Name)
 	}
-	if pod.Status.PodIPs[0].IP != addr.IP {
-		return fmt.Errorf("pod ip doesn't match endpoint ip, skipping: %s vs %s (%s/%s)", pod.Status.PodIPs[0].IP, addr.IP, addr.TargetRef.Namespace, addr.TargetRef.Name)
+	for i := range pod.Status.PodIPs {
```

Comment:
question for golang experts, what is the preferred way to iterate here, by index or by value?
```go
for _, podIP := range pod.Status.PodIPs {
   if podIP.IP == addr.IP {
      ...
```